rust/example-ebpf/src/main.rs

```diff
@@ -4,17 +4,39 @@
 // SPDX-FileCopyrightText: 2024 Benedikt Zinn <benedikt.wh.zinn@gmail.com>
 // SPDX-FileCopyrightText: 2024 Felix Hilgers <felix.hilgers@fau.de>
 // SPDX-FileCopyrightText: 2024 Luca Bretting <luca.bretting@fau.de>
+// SPDX-FileCopyrightText: 2024 Tom Weisshuhn <tom.weisshuhn@fau.de>
 //
 // SPDX-License-Identifier: MIT
 
-use aya_ebpf::{bindings::xdp_action, macros::{map, xdp}, maps::{PerCpuArray, RingBuf}, programs::XdpContext};
+use aya_ebpf::{bindings::xdp_action, macros::{map, xdp, kprobe}, maps::{PerCpuArray, RingBuf}, programs::{XdpContext, ProbeContext}, EbpfContext};
+use example_common::{KProbeData, KProbeTypes};
 
 #[map(name="COUNTER")]
 static PACKET_COUNTER: PerCpuArray<u32> = PerCpuArray::with_max_entries(1, 0);
 
 #[map(name="EVENTS")]
 static EVENTS: RingBuf = RingBuf::with_byte_size(1024, 0);
 
+#[map(name="KProbes")]
+static KPROBES: RingBuf = RingBuf::with_byte_size(1024, 0);
+
+#[kprobe]
+pub fn vfs_write(ctx: ProbeContext) -> Result<(), u32> {
+    let pid = ctx.pid();
+    let tid = ctx.tgid();
+
+    let data = KProbeData{pid, tid, probe_type: KProbeTypes::VfsWrite, ret: false };
+    let mut entry = match KPROBES.reserve::<KProbeData>(0) {
```

Comment:
In the future, when we might send more data, it would be beneficial to directly write them to the entry, instead of allocating them on the stack beforehand. In this case, this is fine.

Reply:
good tip thanks :)
I will change it and pay attention to that in the future!